src/frontend/sharedComponents/PhotoUnpreparedView.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import {View, StyleSheet, Image, ActivityIndicator} from 'react-native';
+
+import {AlertIcon} from './icons';
+import type {PhotoVariant, ViewStyleProp} from '../sharedTypes';
+import {useAttachmentUrlQuery} from '../hooks/server/media';
+import {useObservation} from '../hooks/server/observations.ts';
+
+type Props = {
+  observationId: string;
+  attachmentId: string;
+  variant: PhotoVariant;
+  style?: ViewStyleProp;
+  resizeMode?: 'cover' | 'contain' | 'stretch' | 'center';
+};
+
+const PhotoUnpreparedComponent = ({
+  observationId,
+  attachmentId,
+  variant,
+  resizeMode = 'contain',
+  style,
+}: Props) => {
+  const {
+    data: observation,
+    isError: observationError,
+    isLoading: observationLoading,
+  } = useObservation(observationId);
+  const {
+    data: attachmentUrl,
+    isError: attachmentError,
+    isLoading: attachmentUrlLoading,
+  } = useAttachmentUrlQuery(
+    observation.attachments.find(
+      attachment => attachment.driveDiscoveryId === attachmentId,
+    )!,
+    variant,
+  );
+  const isLoading = observationLoading || attachmentUrlLoading;
+  const isError = observationError || attachmentError;
+
+  return (
+    <View style={[styles.container, style]}>
+      {isLoading ? (
+        <ActivityIndicator />
+      ) : isError ? (
+        <AlertIcon />
+      ) : (
+        <Image
+          // @ts-ignore
+          src={attachmentUrl?.url}
```

Comment:
For some reason it still is `| undefined`, so not sure what's the trick here. Maybe ternary operator is breaking this behavior. I'll leave it as it is if you don't mind, this will never be null with all the checks.

Reply:
[This](https://github.com/digidem/comapeo-mobile/pull/383/commits/b2fe1993d65f50a0c03fbd5f66d6263da2272923) is what i meant. Essentially isPending can replace isLoading.

Reply:
I understood, see comment below. I'm still getting the error. But that could be bug in TS Server on my side.